src/features/DisplayControls/PatientHeader/tests/PatientHeader.spec.jsx

```diff
@@ -1,52 +1,221 @@
-import { render, screen, waitFor } from "@testing-library/react";
+import { fireEvent, render, screen, waitFor } from "@testing-library/react";
 import React from "react";
 import { PatientHeader } from "../components/PatientHeader";
-import { fetchPatientInfo } from "../utils/PatientHeaderUtils";
 
-jest.mock("../utils/PatientHeaderUtils");
+const mockFetchPatientProfile = jest.fn();
+const mockContactDetailsConfig = jest.fn();
+const mockFetchAddressMapping = jest.fn();
+
+jest.mock("../utils/PatientHeaderUtils", () => {
+  const originalModule = jest.requireActual("../utils/PatientHeaderUtils");
+  return {
+    ...originalModule,
+    fetchPatientProfile: () => mockFetchPatientProfile("123"),
+    getConfigsForPatientContactDetails: () => mockContactDetailsConfig(),
+    fetchAddressMapping: () => mockFetchAddressMapping(),
+  };
+});
 
 describe("PatientHeader", () => {
   beforeEach(() => {
-    fetchPatientInfo.mockResolvedValue({
-      person: {
-        preferredName: {
-          display: "John Doe",
+    mockContactDetailsConfig.mockResolvedValue({
+      contactDetails: [
+        "residencePhone",
+        "mobilePhone",
+        "workPhone",
+        "otherPhone",
+        "email",
+      ],
+    });
+    mockFetchPatientProfile.mockResolvedValue({
+      patient: {
```

Comment:
it would be nice if we move these mock values to a different file. test file would look cleaner

Reply:
made the changes